paywall/src/middlewares/providerMiddleware.ts

```diff
@@ -21,17 +46,21 @@ function initializeProvider(
     return
   }
 
-  // provider.enable exists for metamask and other modern dapp wallets and must be called, see:
-  // https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8
-  if (provider.enable) {
-    provider
-      .enable()
-      .then(() => dispatch(providerReady()))
-      .catch(() => dispatch(setError(FATAL_NOT_ENABLED_IN_PROVIDER)))
-  } else {
-    // Default case, provider doesn't have an enable method, so it must already be ready
+  // To avoid flickering when wallet enables without user interaction, delay for a bit
+  // before showing the overlay
+  const timeout = delayedDispatch(dispatch, waitForWallet, 750)
```

Comment:
Not a deal breaker but we might want to consider a better name for this. When looking at line 61 it isnt exactly clear what this might be based on the name